light-base/examples/basic.rs

```diff
@@ -81,32 +78,64 @@ fn main() {
-
-    // The chain is now properly initialized.
+    // Connection to Kusama is now properly initialized.
 
     // `json_rpc_responses` can only be `None` if we had passed `json_rpc: Disabled` in the
     // configuration.
-    let mut json_rpc_responses = json_rpc_responses.unwrap();
+    let mut kusama_json_rpc_responses = kusama_connection.json_rpc_responses.unwrap();
 
-    // Send a JSON-RPC request to the chain.
+    // Ask the client to connect to Statemine.
+    let statemine_connection = client
+        .add_chain(smoldot_light::AddChainConfig {
+            specification: include_str!("../../demo-chain-specs/kusama-statemine.json"),
+            json_rpc: smoldot_light::AddChainConfigJsonRpc::Enabled {
+                max_pending_requests: NonZeroU32::new(128).unwrap(),
+                max_subscriptions: 1024,
+            },
+            // Providing the allocated identifier for Kusama.
+            potential_relay_chains: vec![kusama_connection.chain_id].into_iter(),
+            database_content: "",
+            user_data: (),
+        })
+        .unwrap();
+    // Connection to Statemine is now properly initialized.
+
+    let mut statemine_json_rpc_responses = statemine_connection.json_rpc_responses.unwrap();
+
+    // Send a JSON-RPC request to a chain, Kusama & Statemine in this case.
     // The example here asks the client to send us notifications whenever the new best block has
     // changed.
     // Calling this function only queues the request. It is not processed immediately.
     // An `Err` is returned immediately if and only if the request isn't a proper JSON-RPC request
     // or if the channel of JSON-RPC responses is clogged.
+    //
+    // Kusama:
+    client
+        .json_rpc_request(
+            r#"{"id":1,"jsonrpc":"2.0","method":"chain_subscribeNewHeads","params":[]}"#,
+            kusama_connection.chain_id,
+        )
+        .unwrap();
+    // Statemine:
     client
         .json_rpc_request(
             r#"{"id":1,"jsonrpc":"2.0","method":"chain_subscribeNewHeads","params":[]}"#,
-            chain_id,
+            statemine_connection.chain_id,
         )
         .unwrap();
 
     // Now block the execution forever and print the responses received on the channel of
     // JSON-RPC responses.
     smol::block_on(async move {
         loop {
-            let response = json_rpc_responses.next().await.unwrap();
-            println!("JSON-RPC response: {response}");
+            // The block time of a relay chain is 6 seconds and of a parachain 12 seconds.
+            // As a result, this loop fetches Kusama responses twice for every Statemine response.
+            for _ in 0..2 {
+                let response = kusama_json_rpc_responses.next().await.unwrap();
+                println!("Kusama JSON-RPC response: {response}\n");
+            }
+            let response = statemine_json_rpc_responses.next().await.unwrap();
+            println!("Statemine JSON-RPC response: {response}\n");
```

Comment:
Yeah no, that's not a good practice.

Something like this should work: (requires importing `futures_lite::FutureExt`)

```suggestion
            let (chain_name, response) = async { ("Kusama", kusama_json_rpc_responses.next().await) }.or(async { ("Statemine", statemine_json_rpc_responses.next().await) }).await;
            println!("{chain_name} JSON-RPC response: {response}"); 
```